From: Ashish Arora <ashisharora.linux@outlook.com>
Subject: Re: [PATCH] drm/i915: Discard large BIOS framebuffers causing  display corruption.
Date: Sat, 08 Jan 2022 21:43:18 +1100

On certain 4k panels, the BIOS framebuffer is larger than what panel
requires causing display corruption. Introduce a check for the same.

Signed-off-by: Ashish Arora <ashisharora.linux@outlook.com>
---
 drivers/gpu/drm/i915/display/intel_fbdev.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

Index: lernstick-linux/drivers/gpu/drm/i915/display/intel_fbdev.c
===================================================================
--- lernstick-linux.orig/drivers/gpu/drm/i915/display/intel_fbdev.c	2025-12-05 13:58:06.745683636 +0100
+++ lernstick-linux/drivers/gpu/drm/i915/display/intel_fbdev.c	2025-12-05 13:58:06.737683451 +0100
@@ -224,10 +224,10 @@
 	ifbdev->fb = NULL;
 
 	if (fb &&
-	    (sizes->fb_width > fb->base.width ||
-	     sizes->fb_height > fb->base.height)) {
+	    (sizes->fb_width != fb->base.width ||
+	     sizes->fb_height != fb->base.height)) {
 		drm_dbg_kms(display->drm,
-			    "BIOS fb too small (%dx%d), we require (%dx%d),"
+			    "BIOS fb not valid (%dx%d), we require (%dx%d),"
 			    " releasing it\n",
 			    fb->base.width, fb->base.height,
 			    sizes->fb_width, sizes->fb_height);
